code/game/gamemodes/miniantags/revenant/revenant.dm

```diff
@@ -223,9 +223,7 @@
 	playsound(src, 'sound/effects/screech.ogg', 100, 1)
 	visible_message("<span class='warning'>[src] lets out a waning screech as violet mist swirls around its dissolving body!</span>")
 	icon_state = "revenant_draining"
-	for(var/i = alpha, i > 0, i -= 10)
-		sleep(0.1)
-		alpha = i
+	animate(src, alpha = 0, time = 3 SECONDS)
```

Comment:
animate is async, unlike this previous sleep call. Can you make sure that revs still die correctly with this?

Reply:
I tested it on a clientless revenant by calling the proc directly and then setting its damage high enough to kill it. In both cases, it died successfully.